lib/Event/Model/Asset/ResolveUploadTargetEvent.php

```diff
@@ -38,6 +38,12 @@ public function __construct(int $parentId, string $filename, array $context)
     {
         $this->parentId = $parentId;
         $this->filename = $filename;
+        trigger_deprecation(
+            'pimcore/pimcore',
+            '11.5.0',
+            'The context property is deprecated and will be removed in 12.0.0.
+            Use setArgument() from the ArgumentsAwareTrait instead.'
+        );
         $this->context = $context;
```

Comment:
I cannot comment/suggest on line 37, but I think what you should do instead is removing the `, array $context` parameter and then:

```suggestion
        if (\func_num_args() > 2) {
            trigger_deprecation(
                'pimcore/pimcore',
                '11.5.0',
                'The "$context" argument of "%s()" is deprecated and will be removed in 12.0.0.
                Use "setArgument()" from the "ArgumentsAwareTrait" instead.',
                __METHOD__,
            );
            $this->context = func_get_arg(2);
        }
```

Reply:
Maybe you also need to also default the `$context` property to an empty array in case someone still tries to access it without calling `setArgument()` first.